rpc_component/cli.py

```diff
@@ -242,6 +242,42 @@ def dependency(components_dir, **kwargs):
         )
 
 
+def metadata(components_dir, **kwargs):
+    metadata_dir = kwargs.pop("metadata_dir")
+    metadata_filename = "component_metadata.yml"
+    filepath = os.path.join(metadata_dir, metadata_filename)
+    new_metadata = {"artifacts": [], "dependencies": []}
+    try:
+        metadata = c_lib.load_data(filepath)
+    except yaml.parser.ParserError as e:
+        raise c_lib.ComponentError(
+            "Invalid YAML:"
+            "\n{e}\n".format(
+                e=e
+            )
+        )
+    except FileNotFoundError:
+        metadata = deepcopy(new_metadata)
+
+    subparser = kwargs.pop("metadata_subparser")
+    if subparser == "get":
+        try:
+            return s_lib.component_metadata_schema.validate(metadata)
+        except SchemaError as e:
```

Comment:
I'm not sure this makes it more readable, the schema errors already tend to tell you as much, but if you want to try and improve how these errors present I think it would be better to do it where they are caught at the end of the file (line 506) so everything benefits.

Reply:
Allowing the global exception to do this instead totally makes sense now that you suggest it.